### PR TITLE
Molecule Ring-Based Lifting (Graph to Cell)

### PR DESCRIPTION
**Pull request by bertranMiquel**
Based on [Jin et al. (2018)](https://arxiv.org/abs/1802.04364), this implementation aims to condense a molecule ring into a single element. Utilizing the [QM9 dataset](https://paperswithcode.com/dataset/qm9), a benchmark dataset for molecule prediction, we develop a method to treat rings as 2-cell structures by identifying them through the provided SMILES representations of each molecule. Thus, this submission integrates QM9 dataset into the code.

Here, the elements are the following:
- **Nodes**: Atoms in the molecule.
- **Edges**: Bonds between atoms.
- **2-cells**: Rings in the molecule.

ring

Additionally, attributes inspired by those used in [(Battiloro et al., 2024)](https://arxiv.org/abs/2405.15429) are incorporated into the elements, enhancing the representation of the molecule.
The attributes are:
- **Node**: Atom type, atomic number, and chirality.
- **Edge**: Bond type, conjugation and stereochemistry.
- **Rings**: Ring size, aromaticity, heteroatoms, saturation, hydrophobicity, electrophilicity, nucleophilicity, and polarity.

This pull request is done under the team formed by: Bertran Miquel Oliver, Manel Gil Sorribes, Alexis Molina.

### CODE REVIEW

File: test/transforms/feature_liftings/test_projection_sum.py
Comment (bertranMiquel) [inline test/transforms/feature_liftings/test_projection_sum.py:1]: Deleted due to modification of toy_dataset

File: test/transforms/liftings/graph2cell/test_cycle_lifting.py
Comment (bertranMiquel) [inline test/transforms/liftings/graph2cell/test_cycle_lifting.py:1]: Deleted due to modification of toy_dataset

File: test/transforms/liftings/graph2hypergraph/test_khop_lifting.py
Comment (bertranMiquel) [inline test/transforms/liftings/graph2hypergraph/test_khop_lifting.py:1]: Deleted due to modification of toy_dataset

File: test/transforms/liftings/graph2simplicial/test_clique_lifting.py
Comment (bertranMiquel) [inline test/transforms/liftings/graph2simplicial/test_clique_lifting.py:1]: Deleted due to modification of toy_dataset

**Comment by gbg141** [comment]
Hello @bertranMiquel! Thank you for your submission. As we near the end of the challenge, I am collecting participant info for the purpose of selecting and announcing winners. Please email me (or have one member of your team email me) at guillermo_bernardez@ucsb.edu so I can share access to the voting form. In your email, please include:
 - your first and last name (as well as any other team members)
 - the title of the method you implemented
 - the input domain of the method you implemented
 - the output domain of the method you implemented
 - your pull request number (#14)

Before July 12, make sure that your submission respects all Submission Requirements laid out on [the challenge page](https://pyt-team.github.io/packs/challenge.html). Any submission that fails to meet this criteria will be automatically disqualified.

**Comment by bertranMiquel** [comment]
Hi Guillermo,

Answering to fulfill the submission requirements:
- Name and last name of the participants: Bertran Miquel-Oliver, Manel
Gil-Sorribes, Alexis Molina, Victor Guallar (all of them in cc)
- Title: Molecule Ring-Based Lifting
- Input domain: Graph
- Output domain: Cell Complex
- Pull request: #14
<https://github.com/pyt-team/challenge-icml-2024/pull/14>

Best,
Bertran

Missatge de Guillermo Bernárdez ***@***.***> del dia dl., 8
de jul. 2024 a les 19:11:

**Pull request closed without merging**